client/src/components/profile/components/TimeLine.js

```diff
@@ -68,15 +69,19 @@ class Timeline extends Component {
   }
 
   renderCompletion(completed) {
-    const { idToNameMap } = this.props;
+    const { idToNameMap, idToSlugMap } = this.props;
     const { id, completedDate } = completed;
     const challengeDashedName = idToNameMap[id];
+    let slug = `/challenges/${challengeDashedName}`;
+
+    if (idToSlugMap[id]) {
+      slug = idToSlugMap[id];
+    }
+
```

Comment:
Also, if the challenge id is not found in idToSlugMap, I set the slug to be what it was before so debugging would be easier by seeing the name of the challenge in the url for making and resolving issues.

Let me know if something better comes to mind. 
a flash message probably?